update-data.py

```diff
@@ -105,9 +77,35 @@ def insert_people(cursor, people):
                        ]
                       )
         for stint in person['stints']:
-            logging.debug("  Stint: {}, batch {}, {} - {}".format(
+            batch_id = None
+            batch = stint.get('batch')
+
+            if (batch):
+                batch_id = batch.get('id')
+
+                if (batch_id not in processed_batches and
+                    not person.get('is_faculty') == 'true' and
+                    not stint.get('for_half_batch') == 'true'):
```

Comment:
I think this is incorrectly comparing the boolean `True` with the string `'true'`, and therefore ignoring this condition (because `not (True == 'true') == not False == True`).